ninja-core/src/main/java/ninja/utils/Lambdas.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2016 ninjaframework.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ninja.utils;
+
+import java.lang.invoke.SerializedLambda;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Lambdas {
+    private static final Logger log = LoggerFactory.getLogger(Lambdas.class);
+    
+    /**
+     * The kind of lambda.  Lambdas in Java can either be references to methods
+     * or anonymously defined.
+     * https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html
+     */
+    static public enum Kind {
```

Comment:
I guess the default style in Ninja's codebase is public static... up to you... (in other parts of the Pr, too)...